opennms-config-model/src/main/resources/xsds/eventconf.xsd

```diff
@@ -70,6 +70,12 @@
           </annotation>
         </element>
 
+        <element maxOccurs="1" minOccurs="0" name="priority" type="int">
```

Comment:
1. Should we have a default value?  (such as 100 as suggested in the issue)
2. Maybe make the element name more clear as to what the priority element does?
```XSD
<element maxOccurs="1" minOccurs="0" name="marshaling-priority" type="int" default="100">
```

Reply:
Perhaps better than positiveInteger would be a restriction of 1 - 100.  I'm no expert, so just a guess at this:
```XSD
<element maxOccurs="1" minOccurs="0" name="marshaling-priority" default="100">
  <simpleType>
    <restriction base="positiveInteger">
      <minInclusive value="1"/>
      <maxInclusive value="100"/>
    </restriction>
  </simpleType>
</element>
```

Reply:
There is a default implemented in the Event object and it is '0'.
Using 0 allows us to only prioritize those Events with a non-zero value which helps to reduce processing overhead when matching events.
If we explicitly declare it in the schema however, it will be marshaled out which is not what we want.
I did change the type to 'nonNegativeInteger' and expanded the description.
Please let me know if that is sufficient. Thanks.

Reply:
Hi, did you see the updated code?
This comment is on an 'Outdated' snippet.....
I think it's somethign funky in the gitUI....
The update calls out that 
 - the higher the number - the higher the priority
 - it restricts the valid values to nonNegativeIntegers
see: https://github.com/OpenNMS/opennms/pull/2299/commits/a1769e05808fe21bbccc3a77848dac02e5b18f5b

I don't think we should limit the range - if we place a MAX value on priority and a plugin uses that value, then there will be no way to override the event definition on the file system.
I believe this is the core Use Case for this feature.

I'm not sure that I understand the 3rd question. do you mean making the default in the XSD "0" or something else? could you rephrase? Thanks.

Reply:
The higher the number the higher the priority.  Okay, priority in what exactly?

Reply:
I'm missing the logic on the MAX value and plugin behavior.  But then, I still don't understand from the comments what priority actually means.

Reply:
Yes, I'm asking why doesn't make the default "0" in the XSD any different than treating it as "0" in the code?

Reply:
Priority in matching an incoming event.
As every event comes in and is matched against an event definition, we take the event definitions with non-zero priority and bump them to the front of the queue.
So all of the event definitions with priority > 0, have a chance to match the incoming event before any of the other partitioned event definitions.
This allows the customer to override an event definition that is loaded via a plugin for instance.
With the introduction of Project Sextant, we now load some event definitions via the osgi registry - i.e. they are included in the plugin. 
Without the ability to declare a 'priority' on the event definition on disk, the customer would not be able to override any of the event definitions loaded by the classloader.
The principle Use Case then is for a customer to define a few event definitions with a priority > zero when they need to override the definition included in a plugin.
In the future, there may be more complicated scenarios, e.g. if multiple plugins define the same matching event definition, hence the desire not to cap the priority that the user can apply to their definition on disk.

Reply:
as for making the default "0" in the xsd vs in the code, the difference is that when it is declared in the xsd schema, serializing event definitions will include a 'priority' element even if it wasn't there.
we have many tests that cover the serialization and de-serialization of these objects and they would all break if we changed the schema in this manner.

Reply:
Sorry, one less pang here but I don't want to block the PR.  It seems to me that if the tests would fail then would should fix them since this is a new element in the schema and setting a default is the same has having one specified.  I would hate to make the assumption that a user is only going to specify a priority when it makes sense for a plugin's behavior.  I'm still a fan of putting in a default for any *optional* element or attribute.  My $0.02.

Reply:
Sorry, I phrased that poorly.
It's not because tests would fail that I feel we shouldn't do it that way; It's because forcing the default into the XML serialization would pollute the XML with unnecessary information. The tests failing are a symptom of that.
There is no assumption in the underlying code that customer would only add priority in the Use Case it is addressing - however there would be no benefit for them to do so otherwise, as the code only solves that particular Use Case.
What I'm hoping to achieve is to make the changes to the configuration as simple and clean as possible for the end user. I think one of the challenges with a system as large and as flexible as OpenNMS is to keep things simple.
I feel if this is covered in the Admin Guide properly (*note to self) the way it is implemented in this review will ultimately be the best for the end user.

Reply:
And perhaps if I describe it as follows:
what we are trying to model is Event definitions that can optionally have a priority
those definitions that haven't expressly been assigned a priority, should not have one.
the difference from having a default value for priority vs even having a priority assigned. HTH. :)